addons/mail/static/src/models/message_action_view.js

```diff
@@ -0,0 +1,216 @@
+            switch (this.messageAction.messageActionListOwner) {
+                case this.messageAction.messageActionListOwnerAsDelete:
+                    classNames.push('fa fa-lg fa-trash o_MessageActionView_actionDelete');
+                    break;
+                case this.messageAction.messageActionListOwnerAsEdit:
+                    classNames.push('fa fa-lg fa-pencil o_MessageActionView_actionEdit');
+                    break;
+                case this.messageAction.messageActionListOwnerAsMarkAsRead:
+                    classNames.push('fa fa-lg fa-check o_MessageActionView_actionMarkAsRead');
+                    break;
+                case this.messageAction.messageActionListOwnerAsReaction:
+                    classNames.push('fa fa-lg fa-smile-o o_MessageActionView_actionReaction');
+                    break;
+                case this.messageAction.messageActionListOwnerAsReplyTo:
+                    classNames.push('fa fa-lg fa-reply o_MessageActionView_actionReplyTo');
+                    break;
+                case this.messageAction.messageActionListOwnerAsToggleCompact:
+                    classNames.push('fa fa-lg fa-ellipsis-h o_MessageAction_actionToggleCompact');
+                    break;
+                case this.messageAction.messageActionListOwnerAsToggleStar:
+                    classNames.push('fa fa-lg o_MessageActionView_actionToggleStar');
+                    if (this.messageAction.messageActionListOwner.message.isStarred) {
+                        classNames.push(`fa-star o_MessageActionView_actionToggleStar_active`);
+                    } else {
+                        classNames.push('fa-star-o');
+                    }
+                    break;
+                default:
+                    break;
+            }
+            return classNames.join(' ');
+        },
+        /**
+         * @private
+         * @returns {string}
+         */
+        _computePaddingClassNames() {
+            const isDeviceSmall = this.messaging.device.isSmall;
+            const paddingClassNames = [];
+
+            if (this.messageAction.messageActionListOwner.firstActionView === this) {
+                isDeviceSmall ? paddingClassNames.push('ps-3') : paddingClassNames.push('ps-2');
+            } else {
+                isDeviceSmall ? paddingClassNames.push('ps-2') : paddingClassNames.push('ps-1');
+            }
+
+            if (this.messageAction.messageActionListOwner.lastActionView === this) {
+                isDeviceSmall ? paddingClassNames.push('pe-3') : paddingClassNames.push('pe-2');
+            } else {
+                isDeviceSmall ? paddingClassNames.push('pe-2') : paddingClassNames.push('pe-1');
+            }
+
+            if (isDeviceSmall) {
+                paddingClassNames.push('py-3')
+            } else {
+                paddingClassNames.push('py-2')
+            }
+    
+            return paddingClassNames.join(' ');
```

Comment:
no need for extra new lines, also I don't really like operations being hidden inside `? :` 
just `paddingClassNames.push(condition ? : )` would probably work better